packages/studio/src/components/PageSettingsButton/EntityPageModal.tsx

```diff
@@ -23,13 +24,25 @@ export default function EntityPageModal({
   isOpen,
   handleClose,
 }: PageSettingsModalProps): JSX.Element {
-  const [currGetPathValue, updateGetPathValue, streamScope, updateStreamScope] =
-    useStudioStore((store) => [
-      store.pages.pages[pageName].pagesJS?.getPathValue,
-      store.pages.updateGetPathValue,
-      store.pages.pages[pageName].pagesJS?.streamScope,
-      store.pages.updateStreamScope,
-    ]);
+  const [
+    currGetPathValue,
+    updateGetPathValue,
+    streamScope,
+    updateStreamScope,
+    generateTestData,
+    updateEntityFiles,
+    setActiveEntityFile,
+    refreshActivePageEntities,
+  ] = useStudioStore((store) => [
+    store.pages.pages[pageName].pagesJS?.getPathValue,
+    store.pages.updateGetPathValue,
+    store.pages.pages[pageName].pagesJS?.streamScope,
+    store.pages.updateStreamScope,
+    store.actions.generateTestData,
+    store.pages.updateEntityFiles,
+    store.pages.setActiveEntityFile,
+    store.actions.refreshActivePageEntities,
```

Comment:
Outside the scope of this PR, but I think we should re-examine our Slice design. Specifically, I'm thinking we should have a dedicate Entity Data Slice in the Store. That would be responsible for maintaining an up-to-date mapping of page name to a set of entities. In the `pages` Slice, we'd only have a method for setting the Active Entity used for Previews (`setActiveEntityFile`). I think this could lead to better separation of concerns.

Reply:
the way zustand does slices is a little restrictive, but I think definitely the store can use some reorganizing